RileyLinkKit/ViewControllers/RileyLinkDeviceTableViewController.swift

```diff
@@ -0,0 +1,353 @@
+                cell.textLabel?.text = NSLocalizedString("Pump ID", comment: "The title of the cell showing pump ID")
+                if let pumpID = device.pumpState?.pumpID {
+                    cell.detailTextLabel?.text = pumpID
+                } else {
+                    cell.detailTextLabel?.text = "–"
+                }
+            case .Awake:
+                switch device.pumpState?.awakeUntil {
+                case let until? where until < NSDate():
+                    cell.textLabel?.text = NSLocalizedString("Last Awake", comment: "The title of the cell describing an awake radio")
+                    cell.detailTextLabel?.text = dateFormatter.stringFromDate(until)
+                case let until?:
+                    cell.textLabel?.text = NSLocalizedString("Awake Until", comment: "The title of the cell describing an awake radio")
+                    cell.detailTextLabel?.text = dateFormatter.stringFromDate(until)
+                default:
+                    cell.textLabel?.text = NSLocalizedString("Listening Off", comment: "The title of the cell describing no radio awake data")
+                    cell.detailTextLabel?.text = nil
+                }
+            }
+        case .Commands:
+            switch CommandRow(rawValue: indexPath.row)! {
+            case .Tune:
+                switch (device.radioFrequency, device.lastTuned) {
+                case (let frequency?, let date?):
+                    cell.textLabel?.text = "\(decimalFormatter.stringFromNumber(frequency)!) MHz"
+                    cell.detailTextLabel?.text = dateFormatter.stringFromDate(date)
+                default:
+                    cell.textLabel?.text = NSLocalizedString("Tune radio frequency", comment: "The title of the cell describing the command to re-tune the radio")
+                    cell.detailTextLabel?.text = nil
+                }
+                cell.accessoryType = .DisclosureIndicator
+            case .ChangeTime:
+                cell.textLabel?.text = "Change Time"
+                cell.accessoryType = .DisclosureIndicator
+
+                let localTimeZone = NSTimeZone.defaultTimeZone()
+                let localTimeZoneName = localTimeZone.abbreviation ?? localTimeZone.name
+
+                if let pumpTimeZone = device.pumpState?.timeZone {
+                    let timeZoneDiff = NSTimeInterval(pumpTimeZone.secondsFromGMT - localTimeZone.secondsFromGMT)
+                    let formatter = NSDateComponentsFormatter()
+                    formatter.allowedUnits = [.Hour, .Minute]
+                    let diffString = timeZoneDiff != 0 ? formatter.stringFromTimeInterval(abs(timeZoneDiff)) ?? String(abs(timeZoneDiff)) : ""
+
+                    cell.detailTextLabel?.text = String(format: NSLocalizedString("%1$@%2$@%3$@", comment: "The format string for displaying an offset from a time zone: (1: GMT)(2: -)(3: 4:00)"), localTimeZoneName, timeZoneDiff != 0 ? (timeZoneDiff < 0 ? "-" : "+") : "", diffString)
```

Comment:
Unfortunately `NSDateComponentsFormatter` doesn't have the ability to explicitly set `positiveFormat` and `negativeformat` like `NSNumberFormatter` does, which could clean up the "+"/"-" bit.